### PR TITLE
Add Go module

### PR DESCRIPTION
**Pull request by shuLhan**
The list of required directive is automatically detected by "go mod" from
Gopkg.lock file.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module github.com/github/hub
```
Comment (mislav) [inline go.mod]: Do we need to add `/v2` here?

**Comment by mislav** [comment]
Perhaps we should have a CI build step for Go version 1.11 (and above) that test with `GO111MODULE=on` so that both modules and legacy `vendor/` approach are exercised in tests

File: go.mod (hunk 1 above)
Comment (shuLhan) [reply go.mod]: Since this repository is a program, I don't think its needed.

**Comment by shuLhan** [comment]
Hey @mislav , thank you reviewing.

> Is there anything else that we need to change?

As you already comment above, maybe add test to build without GOPATH in CI.

> How would people use this module support with hub?

Since this repository is already use semver, user that "go get" specific tag will use the go.mod in the commit.

File: go.mod (hunk 1 above)
Comment (mislav) [reply go.mod]: https://github.com/golang/go/wiki/Modules#semantic-import-versioning:

> If the module is version v2 or higher, the major version of the module _must_ be included in both the module path in the go.mod file (e.g., `module example.com/my/mod/v2`) and the package import path (e.g., `import "example.com/my/mod/v2/foo"`).

Comment (shuLhan) [reply go.mod]: Done.

**Comment by mislav** [comment]
This was superseded by https://github.com/github/hub/pull/2138 which doesn't break the CI build.

**Pull request closed without merging**